### ISSUE TITLE
Attention masking in MovingAverageGatedAttention

### ISSUE THREAD

**Issue by jambo6** [facebookresearch/mega#11]
In `moving_average_gated_attention.py` masking in the softmax method is done like
```
if attn_mask is not None:
            qk = qk + attn_mask
```
Is this correct? Surely this does not zero out relevant entries. Do we not want to multiply by the attention mask after application of the softmax?

**Comment by SirRob1997** [comment]
I believe `attn_mask` contains 0‘s and infs that’s why it works as intended.

**Comment by jambo6** [comment]
I see, makes sense. Bit confusing because the element attention uses
```
if attn_mask is not None:
    attn_weights = attn_weights * attn_mask
```
which I assume needs to be of the binary form.

**Comment by mnaylor5** [comment]
I'm working on implementing Mega into Hugging Face's `transformers` library, and I believe you're correct @jambo6. According to the documentation in the `forward` method for `MovingAverageGatedAttention`, `attn_mask` is expected to be a `ByteTensor`, which cannot accept `-inf` as an input. This appears to be a meaningful inconsistency in how this is applied between softmax and element attention -- I haven't yet found where this is happening, but I suspect that a different type of mask is passed depending on the attention method requested (either 0/-inf for additive in softmax or 1/0 for multiplicative in element)

@violet-zct / @jacqueline-he, could you help me understand how this is intended to work?
